app/src/main-process/main.ts

```diff
@@ -189,12 +203,20 @@ function handlePossibleProtocolLauncherArgs(args: ReadonlyArray<string>) {
 
   if (__WIN32__) {
     // Desktop registers it's protocol handler callback on Windows as
-    // `[executable path] --protocol-launcher "%1"`. At launch it checks
-    // for that exact scenario here before doing any processing, and only
-    // processing the first argument. If there's more than 3 args because of a
+    // `[executable path] --protocol-launcher "%1"`. Note that extra command
+    // line arguments might be added by Chromium
+    // (https://electronjs.org/docs/api/app#event-second-instance).
+    // At launch Desktop checks for that exact scenario here before doing any
+    // processing. If there's more than one matching url argument because of a
     // malformed or untrusted url then we bail out.
-    if (args.length === 3 && args[1] === '--protocol-launcher') {
-      handleAppURL(args[2])
+
+    const matchingUrls = args.filter(arg => {
+      const url = URL.parse(arg)
+      return url.protocol && possibleProtocols.has(url.protocol.slice(0, -1))
+    })
+
+    if (args[1] === '--protocol-launcher' && matchingUrls.length === 1) {
```

Comment:
is it reasonable to expect that `--protocol-launcher` is always the second argument?